spacy/matcher/dependencymatcher.pyx

```diff
@@ -122,13 +122,17 @@ cdef class DependencyMatcher:
                     raise ValueError(Errors.E099.format(key=key))
                 visited_nodes[relation["RIGHT_ID"]] = True
             else:
-                if not(
-                    "RIGHT_ID" in relation
-                    and "RIGHT_ATTRS" in relation
-                    and "REL_OP" in relation
-                    and "LEFT_ID" in relation
-                ):
-                    raise ValueError(Errors.E100.format(key=key))
+                required_keys = set(
+                    ("RIGHT_ID", "RIGHT_ATTRS", "REL_OP", "LEFT_ID")
+                )
+                relation_keys = set(relation.keys())
+                missing = required_keys - relation_keys
+                if missing:
+                    missing_txt = ", ".join(list(missing))
+                    raise ValueError(Errors.E100.format(
+                        key=key,
+                        missing=missing_txt
+                    ))
```

Comment:
I like how you've generalized this. Actually, now you can take it one step further and incorporate `required_keys` as a parameter for `E100` instead of having the error code hard-code the required nodes again :-)

Reply:
Great point, added it!